### DISCUSSION TITLE
Question about BAPI_TRANSACTION_COMMIT

### DISCUSSION THREAD

**Discussion by GerritGitHub** [SAP/node-rfc#280] [Q&A] [answered]
Im am testing a BAPI in Node-red using node-red-contrib-saprfc.
I know this discussion is not for node-red-contrib-saprfc but for node-rfc, but because node-red-contrib-saprfc uses node-rfc your answer would maybe be able to help me.

Im using BAPI_GOODSMVT_CREATE to create a goods receipt against a process order and getting a success.
Each time Im running the code a get a new Material Document number as expected which is great.
When I look in SAP however those material documents does not exist, in other words the code did not COMMIT.

This leads me to my question, from what I understand from node-rfc you do NOT have to call BAPI_TRANSACTION_COMMIT.
This is when I read https://github.com/SAP/node-rfc/blob/main/doc/usage.md
"_SAP NWRFC SDK and hence the node-rfc is statefull by default, which makes calling an Update-BAPI or some other RFM that stores intermediate results in the current user’s ABAP session memory very easy. For example, the program just calls the BAPI or function module, and if it succeeds, **just calls BAPI_TRANSACTION_COMMIT on the same connection, and it will run inside the same user session in the backend**_."
From https://github.com/SAP/node-rfc/issues/117 it seems that you need to explicetlly call BAPI_TRANSACTION_COMMIT .

**Comment by bsrdjan** [comment, marked as answer]
Yes, the calling program (node-red) has to call BAPI_TRANSACTION_COMMIT: https://wiki.scn.sap.com/wiki/display/ERPSCM/Goods+Movements+with+BAPI

ABAP example you can replicate / rewrite in node-red: https://wiki.scn.sap.com/wiki/display/ABAP/Sample+code+BAPI_GOODSMVT_CREATE

**Reply by GerritGitHub** [reply]
Thanks [bsrdjan](https://github.com/bsrdjan) for your quick response!!!

**Reply by Alnashar2004** [reply]
how to call BAPI_ALM_NOTIF_CREATE and BAPI_ALM_NOTIF_SAVE in node red in sequence
i called but work indvidually the temp. number created %0000000001 but after save return header all numbers zeros
